src/transaction_builder.js

```diff
@@ -210,8 +212,18 @@ function prepareInput (input, kpPubKey, redeemScript, hashType) {
     // pay-to-scriptHash is not possible without a redeemScript
     if (input.prevOutType === 'scripthash') throw new Error('PrevOutScript is P2SH, missing redeemScript')
 
+    if (!input.pubKeys || input.pubKeys.length === 0) {
+      // maybe we can derive more information about the script
+      // now that we have a kpPubKey?
+      expanded = expandOutput(input.prevOutScript, input.prevOutType, kpPubKey)
+      input.pubKeys = expanded.pubKeys
+      input.signatures = expanded.signatures
```

Comment:
Basically I am never 100% sure what info we have at any point; I am also a little confused on differences between prepareInput/expandInput/expandOutput and at which point are they run (on signing? or at adding?)